Add Cité, Ellsworth Kelly

jack-willturner opened the pull request:
Added directly to `ellsworth_kelly_replications.py`, I hope that's OK? 

If not I'd be happy to separate it out into its own file.

jack-willturner committed:
add cite by Ellsworth Kelly

jack-willturner committed:
add cite to readme

jack-willturner committed:
undo black formatting

jack-willturner committed:
Default line width formatting

sadielbartholomew commented:
Thanks so much for this @jack-willturner. Great idea for a design to replicate and implementation too.

> Added directly to ellsworth_kelly_replications.py, I hope that's OK?

That's perfect, thanks. I think ideally there will be just one script per mini-project, to keep things simple and clean (and if the script gets too complicated then maybe the mini-project should be broken up).

From glancing over the code I see the new code is quite self-contained and shouldn't influence the other outputs, though I am just checking that now by running the script.

Just one question whist I quickly review, how would you like to be attributed: just by your name, your GitHub username as a tag, or both? I would go with 'Contributed by Jack Turner (@jack-willturner)' otherwise, but if you prefer a different variation on your name, say Jack Will Turner, or not touse your username, etc., just let me know. (Alternatively if you prefer I can not add a direct attribution, just in case you wished to remain anonymous ignoring the VCS.)

sadielbartholomew commented:
(Also, just thinking, feel free to add the attribution yourself as an extra commit, then you have full control over it.)

sadielbartholomew commented on ellsworth-kelly-replications/ellsworth_kelly_replications.py:
```diff
@@ -331,8 +351,96 @@ def plot_by_chance(design_choice, plot_four_subplots=True):
+
+    def sample_offset_slopes_thickness():
+        # random offset for starting height
+        y_offset = np.random.normal(0, 0.02, size=1)
```
```suggestion
        y_offset = np.random.normal(0, 0.02)
```

sadielbartholomew commented on ellsworth-kelly-replications/ellsworth_kelly_replications.py:
```diff
@@ -331,8 +351,96 @@ def plot_by_chance(design_choice, plot_four_subplots=True):
+        y_offset = np.random.normal(0, 0.02, size=1)
+
+        # slopes of top and bottom lines of rectangle
+        top_slope = 0.01 + np.random.normal(0, 0.05, size=1)
```
```suggestion
        top_slope = 0.01 + np.random.normal(0, 0.05)
```

sadielbartholomew reviewed (commented):
The code is great and generates a similar (identical except the deliberate randomness) output on my local environment, plus the README table now with an extra output seems like it should render OK.

There was just one small issue, whereby the following warning emerged and it didn't for the same script on `master` (nor did it emerge when `plot_cite(CITE_DESIGN)` was commented out):

> /home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/matplotlib/patches.py:1123: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  xy = np.asarray(xy)

It wasn't a very helpful warning message, but I managed to track it down to the call to `Polygon` where the input data types were a mixture of integers and other arrays which NumPy didn't like (e.g. `[[0, array([-0.01941276])], [0, array([0.06389193])], [1, array([0.16100962])], [1, array([0.04608166])]]`) so if you remove the two `size=1` specifications as suggested, they will become something like `[[0, 0.6460239976374115], [0, 0.8378648815110694], [1, 0.8108640447087803], [1, 0.5714240717955348]]` and the warning goes away. If you could do that, or similar, to address that warning, then I think this is all good to merge. Thanks again!

jack-willturner commented:
Oh good spot - I didn't know `size` behaved like that. 

>  I would go with 'Contributed by Jack Turner (@jack-willturner)'

This sounds good to me! I'll add it in now

jack-willturner committed:
Add attribution to README

Also reshaped Cité picture to be the same as the others (otherwise GitHub seemed to want to move some of the titles underneath the images).

sadielbartholomew committed:
Update ellsworth-kelly-replications to prevent warning

sadielbartholomew commented:
Hi again @jack-willturner, sorry for the delay in getting back to you.

Thanks for updating the PR with the attribution. I committed those suggestions from the code review since it seems you are happy for me to add them directly myself. I'll merge this now as it is ready as far as I can tell.

Thanks once again for your input to the project! I was wondering if you would mind if I tweeted the output from this, essentially to say that I am really happy that the project has had an outside contribution, and such a nice one, and that I will be adding guidelines in the hope to get more from anyone interested in contributing in future? No worries at all if you're not comfortable with that, though. Let me know if it's OK and if so, if you would like me to tag or name you in the tweet.
